### PR TITLE
215612: Deploy Front Door WAF Profile

### PR DESCRIPTION
**Pull request by rajeshkaremane**
# Description
*New Bicep template to deploy Front Door WAF Rule for ADP*

*Link to the relevant work items:  [AB#215612](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/215612)*

# Testing
*Tested locally*

### CODE REVIEW

File: .azuredevops/deploy-platform-core.yaml [hunk 1]
```diff
@@ -124,4 +124,9 @@ extends:
             resourceGroupName: $(servicesResourceGroup)
             privateEndpointDnsRecordsForResources:
               - resourceName: $(infraResourceNamePrefix)$(nc_resource_storageaccount)$(nc_instance_regionid)01
-                resourceGroupName: $(servicesResourceGroup)
\ No newline at end of file
+                resourceGroupName: $(servicesResourceGroup)
+      - name: front_door
+        templates:
+          - name: front-door-waf
```
[the remark this answers is not in the document]
Comment (DaRoszko) [reply .azuredevops/deploy-platform-core.yaml:130]: defra-baseline-waf.parameters etc

**Review by DaRoszko — DISMISSED** [review]

File: infra/front-door/azure-waf.bicep [hunk 2]
```diff
@@ -0,0 +1,91 @@
+@description('Required. The name of the Front Door endpoint to create. This must be globally unique.')
```
Comment (ppotturi) [inline infra/front-door/azure-waf.bicep:1]: is this description correct? It is saying the name of "Front Door endpoint".

Comment (rajeshkaremane) [reply infra/front-door/azure-waf.bicep]: Updated

**Review by DaRoszko — APPROVED** [review]

File: infra/front-door/azure-waf.bicep [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+@description('Optional. The list of exclusions to be configured for Microsoft Bot Manager ruleset  ')
+param microsoftBotManagerRuleSetExclusions array = []
+
+@description('Optional. The PolicySettings for policy.')
+param policySettings object = {
+  enabledState: 'Enabled'
+  mode: 'Prevention'
+  redirectUrl: null
+  customBlockResponseStatusCode: 403
+  customBlockResponseBody: null
```
[the remark this answers is not in the document]
Comment (DaRoszko) [reply infra/front-door/azure-waf.bicep:26]: (custom respose body)

[the remark this answers is not in the document]
Comment (rajeshkaremane) [reply infra/front-door/azure-waf.bicep]: Updated

**Review by DaRoszko — APPROVED** [review]

**Review by ppotturi — APPROVED** [review]

**Pull request merged**